metrics/metricsreporter.go

```diff
@@ -135,29 +135,16 @@ func (m *MetricsReporter) CaptureRoutesPruned(routesPruned uint64) {
 	m.Batcher.BatchAddCounter("routes_pruned", routesPruned)
 }
 
-func (m *MetricsReporter) CaptureRegistryMessage(msg ComponentTagged) {
+func (m *MetricsReporter) CaptureRegistryMessage(msg ComponentTagged, action string) {
 	var componentName string
 	if msg.Component() == "" {
-		componentName = "registry_message"
+		componentName = "registry_message." + action
 	} else {
-		componentName = "registry_message." + msg.Component()
+		componentName = "registry_message." + action + "." + msg.Component()
 	}
 	m.Batcher.BatchIncrementCounter(componentName)
 }
 
-func (m *MetricsReporter) CaptureUnregistryMessage(msg ComponentTagged) {
-	var componentName string
-	if msg.Component() == "" {
-		componentName = "unregistry_message"
-	} else {
-		componentName = "unregistry_message." + msg.Component()
-	}
-	err := m.Sender.IncrementCounter(componentName)
-	if err != nil {
-		m.Logger.Debug("failed-sending-metric", log.ErrAttr(err), slog.String("metric", componentName))
-	}
-}
-
```

Comment:
just for my understanding:

Would you then want to have the "old" total metric and new breakdown metrics in addition?

Reply:
What I mean is that this change:

https://github.com/cloudfoundry/gorouter/pull/456/files#diff-5593a38c8a92809ab9448ee33f120e244e7fc4c0fd0689e239a9e85dc4ae0933L173-R188

will change the emitted metrics for unregister messages from `unregistry_message.*` to `registry_message.*`. This would be a breaking change which doesn't seem to provide much value and could be avoided.

Reply:
You are right. We changed it back.